src/sources/util/mod.rs

```diff
@@ -62,3 +62,24 @@ pub use self::http::HttpSource;
 pub use self::http::HttpSourceAuthConfig;
 #[cfg(any(feature = "sources-aws_sqs", feature = "sources-gcp_pubsub"))]
 pub use self::message_decoding::decode_message;
+
+/// Extract a tag and it's value from input string delimited by a colon character.
+///
+/// Note: the behavior of StatsD if more than one colon is found (which would presumably
+/// be part of the tag value), is to remove any additional colons from the tag value.
+/// Thus Vector expects only one colon character to be present per chunk, so the find()
+/// operation locating the first position is sufficient.
+#[cfg(any(feature = "sources-statsd", feature = "sources-datadog_agent"))]
+pub fn extract_tag_key_and_value(
+    tag_chunk: &str,
```

Comment:
To enable the above, this might need to be `tag_chunk: impl AsRef<str>`

Reply:
Thanks for that assist- I was struggling a bit with that.

Reply:
Addressed in https://github.com/vectordotdev/vector/pull/15639/commits/b16333c736243e040e83950f31d45ab8abce6598